SDK-1204 Updated docs (#686)
box/box-ios-sdk: docs/usage/search.md

To get a list of items matching a search query, call [`client.search.query(query:...)`][search] with the
string to query for.  There are many possible options for advanced search filtering, which can be used to narrow down

> the search results. This method will return an iterator object in the completion you can use to get the results.

Reviewer: ```suggestion
the search results. This method will return an iterator object in the completion, which is used to get the results.
```

To search within metadata for specific values, use the helper methods included in the SDK to construct the metadata
query and call [`client.search.query(query:...)`][search] with the metadata filters and optionally a

> search query.  If no additional content query string is needed, just set the `query` parameter to `nil`. This method will return an iterator object in the completion you can use to get the results. 

Reviewer: ```suggestion
search query.  If no additional content query string is needed, just set the `query` parameter to `nil`. This method will return an iterator object in the completion, which is used to get the results. 
```
